dap-mode.el

```diff
@@ -505,7 +506,7 @@ thread exection but the server will log message."
 (defun dap--parser-read (p output)
   "Parser OUTPUT using parser P."
   (let* ((messages '())
-         (output (string-as-unibyte output))
+         (output (encode-coding-string output 'utf-8 'nocopy))
```

Comment:
No, that function has been there from Emacs 20.3. 
According this thread https://debbugs.gnu.org/cgi/bugreport.cgi?bug=23750, we should not use `string-as-unibyte`, since it means
> Just that `string-as-unibyte` is generally
synonymous with "the author is confused about how coding systems work",
aka "trouble".

Reply:
I saw the same usage of `string-as-unibyte` in `lsp-mode`. We should cleanup there as well